src/main/java/net/minestom/server/network/packet/server/play/TagsPacket.java

```diff
@@ -40,18 +40,18 @@ public TagsPacket() {}
 
     @Override
     public void write(@NotNull BinaryWriter writer) {
-        writeTags(writer, blockTags, name -> Registries.getBlock(name).ordinal());
-        writeTags(writer, itemTags, name -> Registries.getMaterial(name).ordinal());
-        writeTags(writer, fluidTags, name -> Registries.getFluid(name).ordinal());
-        writeTags(writer, entityTags, name -> Registries.getEntityType(name).ordinal());
+        writeTags(writer, blockTags, name -> Registry.BLOCK_REGISTRY.get(name).getNumericalId());
+        writeTags(writer, itemTags, name -> Registry.MATERIAL_REGISTRY.get(name).getNumericalId());
+        writeTags(writer, fluidTags, name -> Registry.FLUID_REGISTRY.get(name).getNumericalId());
+        writeTags(writer, entityTags, name -> Registry.ENTITY_TYPE_REGISTRY.get(name).getNumericalId()); // Not defaulted, so can be null.
     }
 
     @Override
     public void read(@NotNull BinaryReader reader) {
-        readTags(reader, blockTags, id -> NamespaceID.from("minecraft", Block.values()[id].getName()));
-        readTags(reader, itemTags, id -> NamespaceID.from("minecraft", Material.values()[id].getName()));
-        readTags(reader, fluidTags, id -> NamespaceID.from(Fluid.values()[id].getNamespaceID()));
-        readTags(reader, entityTags, id -> NamespaceID.from(EntityType.values()[id].getNamespaceID()));
+        readTags(reader, blockTags, id -> Block.values().get(id).getId());
+        readTags(reader, itemTags, id -> Material.values().get(id).getId());
+        readTags(reader, fluidTags, id -> Fluid.values().get(id).getId());
+        readTags(reader, entityTags, id -> EntityType.values().get(id).getId());
```

Comment:
As before, we should find a better solution than creating a new list each time this function is called.